firefox_ui_tests/remote/security/test_safe_browsing_warning_pages.py

```diff
@@ -0,0 +1,105 @@
+            'https://www.itisatrap.org/firefox/its-an-attack.html'
+            ]
+
+        self.prefs.set_pref('browser.safebrowsing.enabled', True)
+        self.prefs.set_pref('browser.safebrowsing.malware.enabled', True)
+
+        # Give the browser a little time, because SafeBrowsing.jsm takes a
+        # while between start up and adding the example urls to the db.
+        # hg.mozilla.org/mozilla-central/file/46aebcd9481e/browser/base/content/browser.js#l1194
+        time.sleep(3)
+
+    def tearDown(self):
+        try:
+            self.utils.remove_perms('www.itisatrap.org', 'safe-browsing')
+        finally:
+            FirefoxTestCase.tearDown(self)
+
+    def test_warning_pages(self):
+        with self.marionette.using_context("content"):
+            for unsafe_page in self.urls:
+                # Load a test page, then test the get me out button
+                self.marionette.navigate(unsafe_page)
+                # Wait for the DOM to receive events for about:blocked
+                time.sleep(1)
+                self.check_get_me_out_of_here_button(unsafe_page)
+
+                # Load the test page again, then test the report button
+                self.marionette.navigate(unsafe_page)
+                # Wait for the DOM to receive events for about:blocked
+                time.sleep(1)
+                self.check_report_button(unsafe_page)
+
+                # Load the test page again, then test the ignore warning button
+                self.marionette.navigate(unsafe_page)
+                # Wait for the DOM to receive events for about:blocked
+                time.sleep(1)
+                self.check_ignore_warning_button(unsafe_page)
+
+    def check_get_me_out_of_here_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, "getMeOutButton")
+        button.click()
+
+        self.wait_for_condition(lambda mn: self.browser.default_homepage in mn.get_url())
+
+    def check_report_button(self, unsafe_page):
+        button = self.marionette.find_element(By.ID, "reportButton")
+        button.click()
+
+        # Wait for the button to become stale, then wait for page load
+        # so we can verify the url
+        self.wait_for_condition(expected.element_stale(button))
+        # TODO: replace with waitforPageLoad replacement per Bug 1140470
+        self.wait_for_condition(lambda mn: mn.execute_script(
+            "return document.readyState == 'complete'"))
+
+        # Get the base URL to check; this will result in a redirect.
+        with self.marionette.using_context('chrome'):
+            if 'its-a-trap' in unsafe_page:
+                url = self.marionette.execute_script("""
+                   Components.utils.import("resource://gre/modules/Services.jsm");
+                   return Services.urlFormatter.formatURLPref("app.support.baseURL")
+                                                              + "phishing-malware";
+                   """)
```

Comment:
nit: The triple quotes should be aligned with `u` from line 81. Same for the else block.